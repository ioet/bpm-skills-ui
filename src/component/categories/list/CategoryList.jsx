/* eslint-disable react/display-name */
import React from 'react';
import PropTypes from 'prop-types';
import MUIDataTable from 'mui-datatables';
import withStyles from '@material-ui/core/styles/withStyles';
import IconButton from '@material-ui/core/IconButton';
import { Delete } from '@material-ui/icons';
import { CategoryListStyles } from './CategoryListStyles';
import BpmTableCellContainer from '../../bpm-table-cell/BpmTableCellContainer';
import BpmDeleteButtonContainer from '../../bpm-buttons/BpmDeleteButtonContainer';
import { CategoryListConst } from './CategoryListConstants';
import BpmEditButtonContainer from '../../bpm-buttons/BpmEditButtonContainer';
import { editCategory } from '../edit/EditCategoryActions';

const CategoryList = (props) => {
  const {
    classes, categoryList, onRemoveCategories,
  } = props;

  const columns = [
    {
      name: CategoryListConst.COLUMN_ID,
      options: {
        display: 'excluded',
        filter: false,
        sort: false,
        download: false,
      },
    },
    {
      name: CategoryListConst.COLUMN_NAME,
      options: {
        customBodyRender: (value, tableMeta) => (
          <BpmTableCellContainer itemId={tableMeta.rowData[0]}>
            {value}
          </BpmTableCellContainer>
        ),
      },
    },
    {
      name: CategoryListConst.COLUMN_BUSINESS_VALUE,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => (
          <BpmTableCellContainer itemId={tableMeta.rowData[0]}>
            {value}
          </BpmTableCellContainer>
        ),
      },
    },
    {
      name: CategoryListConst.COLUMN_PREDICTIVE_VALUE,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => (
          <BpmTableCellContainer itemId={tableMeta.rowData[0]}>
            {value}
          </BpmTableCellContainer>
        ),
      },
    },
    {
      name: CategoryListConst.COLUMN_EDIT,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => {
          const categoryId = tableMeta.rowData[0];
          return (
            <BpmTableCellContainer itemId={categoryId}>
              <BpmEditButtonContainer
                onEdit={editCategory}
                itemId={categoryId}
              />
            </BpmTableCellContainer>
          );
        },
      },
    },
    {
      name: CategoryListConst.COLUMN_DELETE,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => {
          const categoryId = tableMeta.rowData[0];
          return (
            <BpmTableCellContainer itemId={categoryId}>
              <BpmDeleteButtonContainer
                itemId={categoryId}
              />
            </BpmTableCellContainer>
          );
        },
      },
    },
  ];
  const options = {
    filterType: 'dropdown',
    responsive: 'scroll',
    resizableColumns: false,
    selectableRows: true,
    print: false,
    rowsPerPageOptions: [10, 15, 25],
    customToolbarSelect: selected => (
      <IconButton
        onClick={(e) => {
          e.preventDefault();
          const selectedCategoryIds = selected.data.map(u => categoryList[u.dataIndex][0]);
          onRemoveCategories(selectedCategoryIds);
        }}
        className={classes.iconButton}
      >
        <Delete />
      </IconButton>
    ),
  };

  return (
    <div className={classes.root}>
      <MUIDataTable
        data={categoryList}
        columns={columns}
        options={options}
      />
    </div>
  );
};

CategoryList.propTypes = {
  classes: PropTypes.object.isRequired,
  categoryList: PropTypes.arrayOf(PropTypes.array).isRequired,
  onRemoveCategories: PropTypes.func.isRequired,
};

export default withStyles(CategoryListStyles)(CategoryList);
