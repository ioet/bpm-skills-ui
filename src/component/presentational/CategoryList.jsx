/* eslint-disable react/display-name */
import React from 'react';
import PropTypes from 'prop-types';
import MUIDataTable from 'mui-datatables';
import withStyles from '@material-ui/core/styles/withStyles';
import IconButton from '@material-ui/core/IconButton';
import { Delete } from '@material-ui/icons';
import { CategoryListConst, CategoryListItemConst, Variable } from '../../constants';
import { CategoryListStyles } from '../../styles';
import EditOrPlainTextContainer from '../container/EditOrPlainTextContainer';
import { compareCategoriesByFirstName, getCategoryObjectFromArray } from '../utils/Utils';
import MyTableCellContainer from '../container/MyTableCellContainer';
import MyEditButtonContainer from '../container/MyEditButtonContainer';
import MyDeleteButtonContainer from '../container/MyDeleteButtonContainer';

const CategoryList = (props) => {
  const {
    classes, categoryList, onRemoveCategories,
  } = props;

  const data = [];
  Object.keys(categoryList).forEach((key) => {
    const category = categoryList[key];
    data.push([category.id, category.name, category.business_value, category.predictive_value, false, false]);
    // data.push([category.id, category.name, '2', '2', false, false]);
  });
  data.sort(compareCategoriesByFirstName);

  const columns = [
    {
      name: CategoryListConst.COLUMN_0,
      options: {
        display: 'excluded',
        filter: false,
        sort: false,
        download: false,
      },
    },
    {
      name: CategoryListConst.COLUMN_1,
      options: {
        customBodyRender: (value, tableMeta) => {
          const category = getCategoryObjectFromArray(tableMeta.rowData);
          return (
            <MyTableCellContainer categoryId={category.id}>
              <EditOrPlainTextContainer
                categoryId={category.id}
                value={value}
                name={Variable.NAME}
                label={CategoryListItemConst.EDIT_CATEGORY_NAME}
              />
            </MyTableCellContainer>
          );
        },
      },
    },
    {
      name: CategoryListConst.COLUMN_2,
      options: {
        customBodyRender: (value, tableMeta) => {
          const category = getCategoryObjectFromArray(tableMeta.rowData);
          return (
            <MyTableCellContainer categoryId={category.id}>
              <EditOrPlainTextContainer
                categoryId={category.id}
                value={value}
                name={Variable.BUSINESS_VALUE}
                label={CategoryListItemConst.EDIT_BUSINESS_VALUE}
              />
            </MyTableCellContainer>
          );
        },
      },
    },
    {
      name: CategoryListConst.COLUMN_3,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => {
          const category = getCategoryObjectFromArray(tableMeta.rowData);
          return (
            <MyTableCellContainer categoryId={category.id}>
              <EditOrPlainTextContainer
                categoryId={category.id}
                value={value}
                name={Variable.PREDICTIVE_VALUE}
                label={CategoryListItemConst.EDIT_PREDICTIVE_VALUE}
              />
            </MyTableCellContainer>
          );
        },
      },
    },
    {
      name: CategoryListConst.COLUMN_4,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => {
          const category = getCategoryObjectFromArray(tableMeta.rowData);
          return (
            <MyTableCellContainer categoryId={category.id}>
              <MyEditButtonContainer
                categoryId={category.id}
              />
            </MyTableCellContainer>
          );
        },
      },
    },
    {
      name: CategoryListConst.COLUMN_5,
      options: {
        filter: false,
        sort: false,
        download: false,
        customBodyRender: (value, tableMeta) => {
          const category = getCategoryObjectFromArray(tableMeta.rowData);
          return (
            <MyTableCellContainer categoryId={category.id}>
              <MyDeleteButtonContainer
                categoryId={category.id}
              />
            </MyTableCellContainer>
          );
        },
      },
    },
  ];
  const options = {
    filterType: 'dropdown',
    responsive: 'scroll',
    resizableColumns: true,
    selectableRows: true,
    print: false,
    rowsPerPageOptions: [10, 15, 25],
    customToolbarSelect: selected => (
      <IconButton
        onClick={(e) => {
          e.preventDefault();
          const selectedCategoryIds = selected.data.map(u => data[u.dataIndex][0]);
          onRemoveCategories(selectedCategoryIds);
        }}
        className={classes.iconButton}
      >
        <Delete />
      </IconButton>
    ),
  };

  return (
    <div className={classes.root}>
      <MUIDataTable
        data={data}
        columns={columns}
        options={options}
      />
    </div>
  );
};

CategoryList.propTypes = {
  classes: PropTypes.object.isRequired,
  categoryList: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    business_value: PropTypes.string.isRequired,
    predictive_value: PropTypes.string.isRequired,
  })).isRequired,
  onRemoveCategories: PropTypes.func.isRequired,
};

export default withStyles(CategoryListStyles)(CategoryList);
