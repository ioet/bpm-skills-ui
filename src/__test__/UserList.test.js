import React from 'react';
import CategoryList from '../component/presentational/CategoryList';

describe('<CategoryList />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<CategoryList />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
