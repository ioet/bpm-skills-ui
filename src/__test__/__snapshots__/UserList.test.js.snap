// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> Component Renders correctly 1`] = `
<CategoryList
  classes={
    Object {
      "iconButton": "CategoryList-iconButton-2",
      "root": "CategoryList-root-1",
    }
  }
/>
`;
